### ISSUE TITLE
No printing (bad sensor?)

### ISSUE THREAD

**Issue by lonione** [dukelec/mb#40]
WE have received a new Mbrush cube, which worked fine for a few minutes (maybe 1 hour) a then suddenly stopped. When the button is pressed two short beeps are heard, LEDs turn green but then nothing happens - while moving in either direction, the green LEDs just stay ON. They change back to white after some time (
It looks like the cube is not detecting any movement (maybe?)
The optical sensor is not obstructed, it actually emits light (we checked with a camera)
The wifi and web interface work, we even tried reflashing  the FW (4.33 / 4.16) but it didn't help.
What can we do to identify the problem and fix it? Thanks

**Comment by dukelec** [comment]
It may be that the print surface is too smooth, it is recommended to try a rougher surface.

In addition, there is a Sensor IQC value in the app setting interface, the smaller the value, the better the result. The data will be updated in real time.

**Comment by lonione** [comment]
Thanks for the reply!
I just tried a few different surfaces and nothing - the Sensor IQC value is 255.

**Comment by dukelec** [comment]
It should be the sensor is damaged, please contact the after-sales service to replace the printer.

**Comment by lonione** [comment]
Unfortunately, we got it as a gift and don't know where to go for after-sales support :( 
Is it possible the fault is with a flex cable or some other connection type? Or I could try replacing the sensor if the exact type is known... 
Many thanks!

**Comment by dukelec** [comment]
Yes, there is an FFC cable between the motherboard and the sensor board that you can check to see if there has some poor connection. (The sensor board is a little difficult to disassembly.)

Before disassembly, it is recommended to use the FT232 serial tool, read the motherboard serial port logs, the serial baud rate and pin definition please refer to doc/dev.md, the log will show the sensor information.

Sensor board FFC connector's pin definition, and sensor chip circuit is as follows for your reference:

![Screenshot_2021-06-01_14-36-14](https://user-images.githubusercontent.com/4454365/120278071-6a34a100-c2e7-11eb-9b48-3c2bfd1166dd.png)
![Screenshot_2021-06-01_14-35-37](https://user-images.githubusercontent.com/4454365/120278073-6acd3780-c2e7-11eb-956b-8543062a77f6.png)

If the sensor is really faulty, try blowing the BGA chip with hot air to try to fix a bad solder before replacing it.

**Comment by lonione** [comment]
Thanks @dukelec for the details.
I finally got around to troubleshooting this, and once disassembled the optical sensor basically fell off - not only the transparent plastic housing with the lens, but the chip itself... any chance to source the whole KUDO MB-SB v4.4 somewhere as a spare part?
